src/understand-tidb/plugin.md

```diff
@@ -1 +1,63 @@
 # Plugin
+
+The plugin API allows TiDB to be extended with new features such as audit logging or IP allow/deny listing.
+
+Sample code is provided for a basic audit logging plugin at [`./tidb/plugin/conn_ip_example/`](https://github.com/pingcap/tidb/tree/master/plugin/conn_ip_example). For an example on compiling TiDB and this plugin:
+
+```bash
+plugin="conn_ip_example"
+cd cmd/pluginpkg
+go install
+cd ../../plugin/$plugin
+pluginpkg -pkg-dir . -out-dir .
+cd ../..
+ 
+./bin/tidb-server -plugin-dir /path/to/tidb/plugin/$plugin -plugin-load $plugin-1
+```
+
+An explanation of what this does:
+
+- `cd cmd/pluginpkg` and `go install` compiles the command line utility called `pluginpkg`, which is used to build the plugin.
+- `pluginpkg -pkg-dir . -out-dir .` reads the plugin code + `manifest.toml` file and generates a shared object file for the plugin (`conn_ip_example-1.so`).
+- When the tidb-server starts, it can load plugins in a specified directory (`plugin-dir`).
+
+## Customizing the example plugin
+
+The manifest file describes the capabilities of the plugin, and which features it implements. For a basic version:
+
+```toml
+name = "conn_ip_example"
+kind = "Audit"
+description = "just a test"
+version = "1"
+license = ""
+validate = "Validate"
+onInit = "OnInit"
+onShutdown = "OnShutdown"
+export = [
+    {extPoint="OnGeneralEvent", impl="OnGeneralEvent"},
+    {extPoint="OnConnectionEvent", impl="OnConnectionEvent"}
+]
+```
+
+In addition to this basic example, plugins can also implement an `OnFlush` function. This is called when the statement `FLUSH TIDB PLUGINS pluginName` is executed. TiDB does not require plugins to implement a `OnFlush` function, but when specified it will call this method on all TiDB nodes in the cluster.
```

Comment:
> plugins can also implement an `OnFlush` function

we can provide a type definition of (Audit)Manifest here and setup a link to the source code.

Reply:
I've incorporated this, thanks!